### DISCUSSION TITLE
Attenuation correction artifact

### DISCUSSION THREAD

**Discussion by kirknorth** [ARM-DOE/pyart#1318] [Q&A] [answered]
Processing C-band data from 24-May-2011 revealed a new artifact I had not seen before when performing phase processing and/or attenuation correction. I'll provide code snippets of how I created this later. In the meantime check out the image below. The thin slice in the corrected reflectivity is the artifact I'm referring too, and those values are the missing value -9999.0. I was using `min_ncp = 0.3` and `min_rhv = 0.7`, and using the default values (`min_ncp=0.5` and `min_rhv=0.8`) produces <b>more</b> of these artifacts.

![test](https://cloud.githubusercontent.com/assets/2049028/3434867/b8dcaff0-008e-11e4-89b8-0007554fbb6f.png)

**Comment by josephhardinee** [comment]
Can you add a plot of Phi_dp and K_dp?

**Comment by kirknorth** [comment]
@josephhardinee please see updated image.

**Comment by scollis** [comment]
Hmm.. might be that thin slice of 2nd trip right at the receiver.. Did you hardset the system phase? Or did you let the algorithm try to find it? 

BTW this is one of the issues we hope to address with CMAC2.0

**Comment by josephhardinee** [comment]
Does Kdp go negative in that ray?

**Comment by kirknorth** [comment]
@scollis, here's the phase processing code snippet:

``` python
pdp_corr, kdp_corr = pyart.correct.phase_proc_lp(
                       radar, -2.0, self_const=60000.0, low_z=0.0,
                       high_z=55.0, min_phidp=0.01, min_ncp=0.3, min_rhv=0.7, 
                       fzl=3900.0, overide_sys_phase=False, sys_phase=332.0,
                       nowrap=291, LP_solver='cylp_mp', window_len=35, proc=8,
                       debug=False, really_verbose=False)
```

@josephhardinee, no, please see the updated image for clarity. That being said, there certainly are negative pixels...

**Comment by scollis** [comment]
Hmm.. interesting..

One workaround to keep you going is to set the ray to _FillValue and 
then it will not be used in the gridding.. Is it badvalue from first 
gate? Could you add a rayplot?

**Comment by kirknorth** [comment, marked as answer]
@scollis, yup it's `_FillValue` for the entire ray(s). The May 24th case suffered from second trip echoes much more than the other cases I've processed, so my first guess is these artifacts are linked to that. That being said, these artifacts decrease the more I lower `ncp_min` towards 0, i.e. the more I allow second trip pixels in. Furthermore, I see this artifact on several tilts at different azimuths.

![test](https://cloud.githubusercontent.com/assets/2049028/3435047/dd31e4ae-0090-11e4-847d-7f72ed13a389.png)
